scripts/setup_haskell.hs

```diff
@@ -0,0 +1,355 @@
+  --resolver lts-8.14
+  --package aeson
+  --package ansi-terminal
+  --package foldl
+  --package mtl
+  --package stache
+  --package system-filepath
+  --package text
+  --package transformers
+  --package turtle
+-}
+
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE QuasiQuotes #-}
+{-# LANGUAGE RecordWildCards #-}
+
+import Control.Applicative ((<|>), empty)
+import Control.Exception (bracket_)
+import qualified Control.Foldl as Foldl
+import Control.Monad (mfilter, unless, when)
+import Control.Monad.IO.Class (MonadIO, liftIO)
+import Control.Monad.Reader (MonadReader, ask, runReaderT)
+import Control.Monad.Trans.Maybe (MaybeT(..), runMaybeT)
+import Data.Aeson ((.=), object)
+import Data.Foldable (forM_)
+import Data.Maybe (isJust, listToMaybe)
+import Data.Monoid ((<>))
+import qualified Data.Text as Text
+import Data.Text (Text)
+import qualified Data.Text.IO as Text.IO
+import Data.Text.Lazy (toStrict)
+import Filesystem.Path.CurrentOS (FilePath, (</>))
+import qualified Filesystem.Path.CurrentOS as FS
+import Prelude hiding (FilePath)
+import qualified System.Console.ANSI as ANSI
+import qualified System.IO
+import System.Info (os)
+import Text.Mustache (Template, renderMustache)
+import Text.Mustache.Compile.TH (mustache)
+import qualified Turtle
+
+data HvnConfig = HvnConfig
+  { hvnCfgHome           :: FilePath
+  , hvnCfgDest           :: FilePath
+  , hvnCfgHoogleDb       :: Bool
+  , hvnCfgHelperBinaries :: Bool
+  } deriving (Show)
+
+data HvnArgs = HvnArgs
+  { hvnArgsNoHoogleDb       :: Bool
+  , hvnArgsNoHelperBinaries :: Bool
+  } deriving (Show)
+
+main :: IO ()
+main = do
+  HvnArgs{..} <- Turtle.options "Haskell Vim Now - setup Haskell specifics"
+                    cliParser
+  print HvnArgs{..}
+  hvnCfgHome <- hvnHomeDir
+  let hvnCfgDest = hvnCfgHome </> "haskell-vim-now"
+      hvnCfgHoogleDb = not hvnArgsNoHoogleDb
+      hvnCfgHelperBinaries = not hvnArgsNoHelperBinaries
+  runReaderT setup HvnConfig{..}
+  Turtle.exit Turtle.ExitSuccess
+
+cliParser :: Turtle.Parser HvnArgs
+cliParser = HvnArgs
+        <$> Turtle.switch "no-hoogle" 'g'
+              "Disable Hoogle database generation."
+        <*> Turtle.switch "no-helper-bins" 'b'
+              "Disable install of helper binaries (mainly for CI)."
+
+hvnHomeDir :: (MonadIO m) => m FilePath
+hvnHomeDir = do
+  mXdgConfigHome <- Turtle.need "XDG_CONFIG_HOME"
+  maybe
+    (fmap (</> ".config") Turtle.home)
+    (pure . textToFilePath)
+    mXdgConfigHome
+
+setup :: (MonadIO m, MonadReader HvnConfig m) => m ()
+setup = do
+  setupHaskell
+  msg "HASKELL VIM NOW install of Haskell specifics successfully finished"
+
+setupHaskell :: (MonadIO m, MonadReader HvnConfig m) => m ()
+setupHaskell = do
+  HvnConfig{..} <- ask
+  hasStack <- hasExecutable "stack"
+  unless hasStack $ do
+    err "Installer requires Stack - installation instructions:"
+    msg "  http://docs.haskellstack.org/en/stable/README/#how-to-install"
```

Comment:
Is it possible to have this `unless hasStack` terminate the `setuphaskell` function? That way you could remove `when hasStack` underneath and un-indent that whole big section by one level.